test/mongodb.test.js

```diff
@@ -4,6 +4,36 @@ var should = require('./init.js');
 
 var Superhero, User, Post, PostWithStringId, db;
 
+describe('lazyConnect', function() {
+  it('should skip connect phase (lazyConnect = true)', function(done) {
+    var ds = getDataSource({
+      host: '127.0.0.1',
+      port: 4,
+      lazyConnect: true,
+    });
+    var errTimeout = setTimeout(function() {
+      done();
+    }, 5000);
+
+    ds.on('error', function(err) {
+      clearTimeout(errTimeout);
```

Comment:
@bajtos actually the test case would fail after 5s delay, if error event fired, but `done(err)` will speed the failure of test, which makes test fail immediately when error fired.
Will add it then.